src/node_report.cc

```diff
@@ -0,0 +1,790 @@
+ *   Trigger signal selection
+ *   NodeReport filename
+ *   NodeReport directory
+ *   Verbose mode
+ ******************************************************************************/
+unsigned int ProcessNodeReportEvents(const char *args) {
+  // Parse the supplied event types
+  unsigned int event_flags = 0;
+  const char* cursor = args;
+  while (*cursor != '\0') {
+    if (!strncmp(cursor, "exception", sizeof("exception") - 1)) {
+      event_flags |= NR_EXCEPTION;
+      cursor += sizeof("exception") - 1;
+    } else if (!strncmp(cursor, "fatalerror", sizeof("fatalerror") - 1)) {
+       event_flags |= NR_FATALERROR;
+       cursor += sizeof("fatalerror") - 1;
+    } else if (!strncmp(cursor, "signal", sizeof("signal") - 1)) {
+        event_flags |= NR_SIGNAL;
+        cursor += sizeof("signal") - 1;
+    } else {
+      fprintf(stderr, "Unrecognised argument for nodereport events option: %s\n", cursor);
+       return 0;
+    }
+    if (*cursor == '+') {
+      cursor++;  // Hop over the '+' separator
+    }
+  }
+  return event_flags;
+}
+
+unsigned int ProcessNodeReportCoreSwitch(const char *args) {
+  if (strlen(args) == 0) {
+    fprintf(stderr, "Missing argument for nodereport core switch option\n");
+    return 0;
+  }
+  // Parse the supplied switch
+  if (!strncmp(args, "yes", sizeof("yes") - 1) || !strncmp(args, "true", sizeof("true") - 1)) {
+    return 1;
+  } else if (!strncmp(args, "no", sizeof("no") - 1) || !strncmp(args, "false", sizeof("false") - 1)) {
+    return 0;
+  } else {
+    fprintf(stderr, "Unrecognised argument for nodereport core switch option: %s\n", args);
+  }
+  return 1;  // Default is to produce core dumps
+}
+
+unsigned int ProcessNodeReportSignal(const char *args) {
+#ifdef _WIN32
+  return 0; // no-op on Windows
+#else
+  if (strlen(args) == 0) {
+    fprintf(stderr, "Missing argument for nodereport signal option\n");
+    return 0;
+  }
+  // Parse the supplied switch
+  if (!strncmp(args, "SIGUSR2", sizeof("SIGUSR2") - 1)) {
+    return SIGUSR2;
+  } else if (!strncmp(args, "SIGQUIT", sizeof("SIGQUIT") - 1)) {
+    return SIGQUIT;
+  } else {
+    fprintf(stderr, "Unrecognised argument for nodereport signal option: %s\n", args);
+  }
+  return SIGUSR2;  // Default is SIGUSR2
+#endif
+}
+
+void ProcessNodeReportFileName(const char *args) {
+  if (strlen(args) == 0) {
+    fprintf(stderr, "Missing argument for nodereport filename option\n");
+    return;
+  }
+  if (strlen(args) > NR_MAXNAME) {
+    fprintf(stderr, "Supplied nodereport filename too long (max 48 characters)\n");
+    return;
+  }
+  strcpy(report_filename, args);
+}
+
+void ProcessNodeReportDirectory(const char *args) {
+  if (strlen(args) == 0) {
+    fprintf(stderr, "Missing argument for nodereport directory option\n");
+    return;
+  }
+  if (strlen(args) > NR_MAXPATH) {
+    fprintf(stderr, "Supplied nodereport directory path too long (max 2048 characters)\n");
+    return;
+  }
+  strcpy(report_directory, args);
```

Comment:
It's not exactly wrong but `strcpy()` will set off most static analyzers.  I'd use `snprintf(report_directory, sizeof(report_directory), "%s", args)`.